ui/ui-frontend/projects/archive-search/src/app/archive/archive-search/archive-search.component.spec.ts

```diff
@@ -252,6 +253,38 @@ describe('ArchiveSearchComponent', () => {
         expect(elementBtn.length).toBe(2);
       });
     });
+
+    describe('checkChildrenBoxChange', () => {
+      it('should include the unselected child when parent is checked, into the list listOfUAIdToExclude', () => {
+        component.isAllChecked = true;
+        const event = {
+          stopPropagation: () => {},
+          target: {
+            checked: false,
+          },
+        } as unknown as Event;
+        const preventDefaultSpy = spyOn(event, 'stopPropagation');
+        const unit = { '#id': '1234', '#unitups': [''], '#allunitups': [''], '#unitType': UnitType.HOLDING_UNIT, '#opi': '1234' };
+        component.checkChildrenBoxChange(unit, event);
+        expect(component.listOfUAIdToExclude.length === 1).toBeTruthy();
+        expect(preventDefaultSpy).toHaveBeenCalled();
```

Comment:
Tu peux simplifier :
```suggestion
        const event: Event = jasmine.createSpyObj<Event>(['stopPropagation'], { target: { checked: false } as HTMLInputElement });
        const unit = { '#id': '1234', '#unitups': [''], '#allunitups': [''], '#unitType': UnitType.HOLDING_UNIT, '#opi': '1234' };
        component.checkChildrenBoxChange(unit, event);
        expect(component.listOfUAIdToExclude.length === 1).toBeTruthy();
        expect(event.stopPropagation).toHaveBeenCalled();
```

Reply:
Done